backends/gstreamer/lib.rs

```diff
@@ -152,3 +188,248 @@ impl<'a> Drop for MutexGuard<'a> {
         }
     }
 }
+
+struct GStreamerRegistryScanner {
+    is_initialized: bool,
+    supported_mime_types: HashSet<&'static str>,
+    supported_codecs: HashSet<&'static str>,
+}
+
+impl GStreamerRegistryScanner {
+    pub fn new() -> GStreamerRegistryScanner {
+        GStreamerRegistryScanner {
+            is_initialized: false,
+            supported_mime_types: HashSet::new(),
+            supported_codecs: HashSet::new(),
+        }
+    }
+
+    pub fn is_container_type_supported(&mut self, container_type: &str) -> bool {
+        if !self.is_initialized {
+            self.initialize();
+            self.is_initialized = true;
+        }
```

Comment:
what about using https://doc.rust-lang.org/std/sync/struct.Once.html for this, and a macro to call that once structure? :)

Reply:
`lazy_static` takes care of this :)